Find a functionally equivalent code:android.appwidget.AppWidgetProvider.onReceive:COMMENT
Method Modifier: public      
Comment:// BEGIN_INCLUDE(onReceive)

@@ -27,31 +27,41 @@
         this.onEnabled(context);
     } else if (AppWidgetManager.ACTION_APPWIDGET_DISABLED.equals(action)) {
         this.onDisabled(context);
+    } else if (AppWidgetManager.ACTION_APPWIDGET_RESTORED.equals(action)) {
+        Bundle extras = intent.getExtras();
+        if (extras != null) {
+            int[] oldIds = extras.getIntArray(AppWidgetManager.EXTRA_APPWIDGET_OLD_IDS);
+            int[] newIds = extras.getIntArray(AppWidgetManager.EXTRA_APPWIDGET_IDS);
+            if (oldIds != null && oldIds.length > 0) {
+                this.onRestored(context, oldIds, newIds);
+                this.onUpdate(context, AppWidgetManager.getInstance(context), newIds);
+            }
+        }
     }
 }

